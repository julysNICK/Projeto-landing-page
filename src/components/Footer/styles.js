import styled, { css } from 'styled-components';

export const Container = styled.footer`
  ${({ theme }) => css`
    text-align: center;
    a {
      color: inherit;
      text-decoration: none;
      font-size: ${theme.font.sizes.small};
    }
  `}
`;
