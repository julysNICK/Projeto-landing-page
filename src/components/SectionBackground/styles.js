import styled, { css } from 'styled-components';

const containerBackgroundActive = (theme) => css`
  background: ${theme.colors.primaryColor};
  color: ${theme.colors.white};
`;

export const Container = styled.div`
  ${({ theme, background }) => css`
    background: ${theme.colors.white};
    color: ${theme.colors.primaryColor};
    ${background && containerBackgroundActive(theme)};
    min-height: 100vh;
    display: flex;
    align-items: center;
  `}
`;
