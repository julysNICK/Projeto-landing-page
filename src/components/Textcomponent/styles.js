import styled, { css } from 'styled-components';

export const Container = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.medium};
  `}
`;
