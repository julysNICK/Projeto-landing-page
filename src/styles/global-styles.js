import { createGlobalStyle, css } from 'styled-components';

export const GlobalStyles = createGlobalStyle`

*{
  margin:0;
  padding:0;
  box-sizing:border-box;
}
html{
  font-size:62.5;
}
body{
 font-size:1.6rem;
 font-family:${({ theme }) => theme.font.family.default};
}

h1,h2,h3,h4,h5,h6{
  font-family:${({ theme }) => theme.font.family.secondary};
}
`;
