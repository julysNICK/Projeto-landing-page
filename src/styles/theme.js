export const theme = {
  colors: {
    primaryColor: '#0A1128',
    secondaryColor: '#dc143c',
    white: '#FFFFFF',
  },
};
