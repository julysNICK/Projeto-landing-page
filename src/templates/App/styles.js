import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ background }) => css`
    background: ${background};
  `}
`;
