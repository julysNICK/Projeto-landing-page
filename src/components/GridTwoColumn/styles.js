import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme, background }) => css`
    display: grid;
    grid-template-columns: 1fr 2fr;
    align-items: center;
    gap: ${theme.spacings.large};
    @media ${theme.font.media.lteMedium} {
      grid-template-columns: 1fr;
      text-align: center;
    }
    ${Title} {
      @media ${theme.font.media.lteMedium};
      color: ${background ? theme.colors.white : theme.colors.primaryColor};
    }
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    @media ${theme.font.media.lteMedium} {
      margin-bottom: ${theme.spacings.large};
    }
  `}
`;

export const ImageContainer = styled.div`
  ${({ theme }) => css``}
`;

export const Image = styled.img`
  ${({ theme }) => css``}
`;
