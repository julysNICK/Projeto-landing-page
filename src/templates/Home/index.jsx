import { useEffect, useState } from 'react';

import * as Styled from './styles';

import { mapData } from '../../api/map-data';

import { Heading } from '../../components/Heading';
import { GridTwoColum } from '../../components/GridTwoColumn';
import { GridContent } from '../../components/GridContent';
import { GridText } from '../../components/GridText';
import { GridImage } from '../../components/GridImage';

import { mockBase } from '../Base/mock';
import { Base } from '../Base';
import { PageNotFound } from '../PageNotFound';
import { Loading } from '../Loading';

function Home() {
  const [data, setData] = useState([]);
  useEffect(() => {
    const load = async () => {
      try {
        const data = await fetch(
          'http://localhost:1337/pages/?slug=landing-page',
        );
        const json = await data.json();
        const pageData = mapData(json);
        setData(pageData[0]);
      } catch (e) {
        console.log(e);
        setData(undefined);
      }
    };
    load();
  }, []);
  if (data === undefined) {
    return <PageNotFound />;
  }
  if (data && !data.slug) {
    return <Loading />;
  }

  const { menu, sections, footerHtml, slug } = data;
  const { links, text, link, srcImg } = menu;

  return (
    <Base
      links={links}
      footerHtml={footerHtml}
      logoData={{ text, link, srcImg }}
    >
      {sections.map((section, index) => {
        const { component } = section;
        const key = `${slug}-${index}`;

        if (component === 'section.section-two-columns') {
          return <GridTwoColum key={key} {...section} />;
        }

        if (component === 'section.section-content') {
          return <GridContent key={key} {...section} />;
        }

        if (component === 'section.section-grid-text') {
          return <GridText key={key} {...section} />;
        }

        if (component === 'section.section-grid-image') {
          return <GridImage key={key} {...section} />;
        }
      })}
    </Base>
  );
}

export default Home;
