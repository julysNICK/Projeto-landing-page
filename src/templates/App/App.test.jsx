import { render, screen } from '@testing-library/react';

import Home from '.';
import { renderTheme } from '../../styles/render-theme';
import { theme } from '../../styles/theme';

test('renders learn react link', () => {
  const { debug } = renderTheme(<Home />);

  const headingContainer = screen.getByRole('heading', { name: 'hello' })
    .parentElement;
  expect(headingContainer).toHaveStyle({
    background: theme.colors.mainBg,
  });

  expect(headingContainer).toMatchSnapshot();

  expect(headingContainer).toHaveStyleRule('background', 'red');
});
