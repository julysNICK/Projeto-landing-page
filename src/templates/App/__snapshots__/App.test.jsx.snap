// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
.c0 {
  background: red;
}

<div
  class="c0"
>
  <h1>
    hello
  </h1>
</div>
`;
